conda_repo.rb

```diff
@@ -0,0 +1,63 @@
+# frozen_string_literal: true
+
+require "msgpack"
+require "json"
+require "httparty"
+require "net/http"
+
+class CondaRepo
+  CHANNEL_PATH = File.dirname(__FILE__) + "/tmp/channels"
+  CHANNELS = [
+    # channel, domain, directory_path_to channeldata.json
+    ["pkgs/main", "repo.anaconda.com"],
+    # ["conda-forge", "conda.anaconda.org", "/conda-forge"],
+  ].freeze
+
+  def initialize(redis)
+    @redis = redis
+  end
+
+  def package_names
+    @redis.smembers("package_names")
+  end
+
+  def package(channel, name)
+    pack = @redis.get("packages:#{channel}/#{name}")
+    return nil unless pack
+
+    MessagePack.unpack(pack.force_encoding("ASCII-8BIT"))
+  end
+
+  def download_and_parse_packages
+    channels_with_packages = {}
+    CHANNELS.each do |channel, domain|
+      channel_packages = download_json(channel, domain)["packages"]
+      channels_with_packages[channel] ||= {}
+      channel_packages.each do |package_name, package|
+        channels_with_packages[channel][package_name] = package
+      end
+    end
+
+    channels_with_packages
+  end
+
+  def update_packages
+    download_and_parse_packages.each do |channel, packages|
+      @redis.sadd("package_names", packages.keys.map { |name| "#{channel}/#{name}" })
+
+      packages.each_with_index do |(name, package_info), index|
+        version = package_info["version"]
+        key = "#{channel}/#{name}"
+        @redis.set("packages:#{key}", MessagePack.pack(version))
+        GC.start if index % 100 == 0
```

Comment:
This makes me nervous - do we really know better than the GC does when it should run?